src/Mvc/Mvc.Core/src/Formatters/InputFormatter.cs

```diff
@@ -159,7 +159,7 @@ public virtual IReadOnlyList<string> GetSupportedContentTypes(string contentType
                     {
                         if (mediaTypes == null)
                         {
-                            mediaTypes = new List<string>();
+                            mediaTypes = new List<string>(SupportedMediaTypes.Count);
```

Comment:
I'd change these and the following in the next files to the 
```c#
mediaTypes ??= new List<string>(/* ... */)
```
pattern when you're on it.